### PR TITLE
fix(shaders/directx): replace modulus with bitwise operator

### PR DESCRIPTION
**Pull request by iMakeSoftware**
Replace modulo operation with bitwise operation to optimize performance and remove warning messages on some GPUs

### CODE REVIEW

File: src_assets/windows/assets/shaders/directx/include/base_vs.hlsl [hunk 1]
```diff
@@ -25,7 +25,7 @@ vertex_t generate_fullscreen_triangle_vertex(uint vertex_id, float2 subsample_of
         float2 rotation_center = { 0.5, 0.5 };
         tex_coord = round(rotation_center + mul(rotation_matrix, tex_coord - rotation_center));
```
Comment (FrogTheFrog) [inline src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:27]:
```suggestion

        // Swap the xy offset coordinates if the texture is rotated an odd number of times.
```

**Review by FrogTheFrog — APPROVED** [review]
A neat optimization.

I would suggest adding a comment on what's happening at least (even if I have no idea what's happening).

File: src_assets/windows/assets/shaders/directx/include/base_vs.hlsl (hunk 1 above)
Comment (iMakeSoftware) [reply src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:27]: Myself and a number of others from the logs I have seen received these warnings during sunshine startup during shader compilation:

#####
Info: Compiling shaders...
Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
Info: System tray created
Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
Info: Compiled shaders
####

This is due to the fact most GPU hardware (Nvidia and at least some Intel GPUs are less efficient at performing division operations and do not compile the source code for integers or float values to optimal lower level code. I'm not sure if the difference will be noticeable other than decluttering the logs but it's a 2 character change that even if it's not technically necessary, is an improvement.

Comment (FrogTheFrog) [reply src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:27]: Oh, I'm fine with this and I have seen the log myself.

My only suggestion is to add a comment explaining that this is a quick "is odd" check for people that are unfamiliar with bitwise operations.

Comment (iMakeSoftware) [inline src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:27]: /*bitwise operator used as modulus operation on integer does not compile optimally and generates warnings, bitwise operation "& 1" is equivalent to "% 2" in a form that results in optimized lower level code*/

Comment (iMakeSoftware) [reply src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:27]: Ah yeh sorry it's not exactly self explanatory 😅, I've added a comment for future reference

Comment (ReenigneArcher) [reply src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:27]: @iMakeSoftware I think there might be a miscommunication somewhere. The suggestion is to add a comment to the code, not to the PR.

Comment (iMakeSoftware) [reply src_assets/windows/assets/shaders/directx/include/base_vs.hlsl]: No, not a miscommunication the second time... just a horrible UI in the GitHub mobile app, please see latest commit with comment added on line 27 as per suggestion

File: src_assets/windows/assets/shaders/directx/include/base_vs.hlsl [hunk 2]
```diff
@@ -24,8 +24,11 @@ vertex_t generate_fullscreen_triangle_vertex(uint vertex_id, float2 subsample_of
                                      sin(rotation_radians), cos(rotation_radians) };
         float2 rotation_center = { 0.5, 0.5 };
         tex_coord = round(rotation_center + mul(rotation_matrix, tex_coord - rotation_center));
-
-        if (rotate_texture_steps % 2) {
+/* 
+bitwise operation used in place of modulus operation as mod operator on integer does not compile optimally generating warnings, 
+bitwise operation "& 1" is equivalent to "% 2" in a form that results in optimized lower level code.
+*/
+        if (rotate_texture_steps & 1) {
```
Comment (ReenigneArcher) [inline src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:31]:
```suggestion
        // Swap the xy offset coordinates if the texture is rotated an odd number of times.
        if (rotate_texture_steps & 1) {

```

File: src_assets/windows/assets/shaders/directx/include/base_vs.hlsl [hunk 3]
```diff
@@ -24,8 +24,9 @@ vertex_t generate_fullscreen_triangle_vertex(uint vertex_id, float2 subsample_of
                                      sin(rotation_radians), cos(rotation_radians) };
         float2 rotation_center = { 0.5, 0.5 };
         tex_coord = round(rotation_center + mul(rotation_matrix, tex_coord - rotation_center));
+        // Swap the xy offset coordinates if the texture is rotated an odd number of times.
+        if (rotate_texture_steps & 1) {
```
Comment (ReenigneArcher) [inline src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:28]:
```suggestion

        // Swap the xy offset coordinates if the texture is rotated an odd number of times.
        if (rotate_texture_steps & 1) {

```

File: src_assets/windows/assets/shaders/directx/include/base_vs.hlsl [hunk 4]
```diff
@@ -25,7 +25,10 @@ vertex_t generate_fullscreen_triangle_vertex(uint vertex_id, float2 subsample_of
         float2 rotation_center = { 0.5, 0.5 };
         tex_coord = round(rotation_center + mul(rotation_matrix, tex_coord - rotation_center));
 
-        if (rotate_texture_steps % 2) {
+        // Swap the xy offset coordinates if the texture is rotated an odd number of times.
+        if (rotate_texture_steps & 1) {
+
```
Comment (ReenigneArcher) [inline src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:30]:
```suggestion

```

File: src_assets/windows/assets/shaders/directx/include/base_vs.hlsl [hunk 5]
```diff
@@ -25,7 +25,9 @@ vertex_t generate_fullscreen_triangle_vertex(uint vertex_id, float2 subsample_of
...
```
Comment (ReenigneArcher) [inline src_assets/windows/assets/shaders/directx/include/base_vs.hlsl:30]:
```suggestion
```

**Comment by ReenigneArcher** [comment]
Sorry for the spam, trying to clean this up from GitHub ui.

**Pull request merged**